src/Tooling/Bit.Tooling.Templates/TodoTemplate/Api/Middlewares/HttpResponseExceptionHandlerMiddleware.cs

```diff
@@ -0,0 +1,58 @@
+using System.Net;
+using System.Reflection;
+
+namespace TodoTemplate.Api.Middlewares;
+
+public class HttpResponseExceptionHandlerMiddleware
+{
+    private readonly RequestDelegate _next;
+
+    public HttpResponseExceptionHandlerMiddleware(RequestDelegate next)
+    {
+        _next = next;
+    }
+
+    public async Task Invoke(HttpContext context, IHostEnvironment webHostEnvironment)
+    {
+        context.Response.Headers.Add("Request-ID", context.TraceIdentifier);
+
+        try
+        {
+            await _next(context);
+        }
+        catch (Exception e)
+        {
+            var exception = UnWrapException(e);
+            var isKnownException = exception is KnownException;
+            var statusCode = (int)(exception is RestException restExp ? restExp.StatusCode : HttpStatusCode.InternalServerError);
+            
+            var restExceptionPayload = RestExceptionPayloadBuilder(webHostEnvironment, isKnownException, exception);
+
+            context.Response.StatusCode = statusCode;
+            await context.Response.WriteAsJsonAsync(restExceptionPayload);
+        }
+    }
+
+    private RestExceptionPayload RestExceptionPayloadBuilder(IHostEnvironment webHostEnvironment, bool isKnownException, Exception exception)
```

Comment:
please rename this method to `CreateRestExceptionPayload`.
this name is not a suitable name for a method. it's rather a class name.